controllers/front/HistoryController.php

```diff
@@ -87,7 +87,11 @@ public function getTemplateVarOrders(): array
      */
     public static function getUrlToInvoice(Order $order, Context $context)
     {
-        $url_to_invoice = '';
+        $url_to_invoice = Hook::exec('actionGetUrlInvoice', ['order' => $order]);
+
+        if (Validate::isUrl($url_to_invoice)) {
```

Comment:
I like the idea!

One suggestion: Since the method didn't validate the URL before, having it like this is a breaking change. You should check whether `$customUrlToInvoice` (that's how I suggest naming the variable) is not empty and only then validate the URL. Otherwise, the method should behave like before.

Ok, two more things :D

Have you checked if there are more places to retrieve the URL? Wouldn't it be wise to have this replacement mechanism globally?

And, last but not least, you can send this pull request on `9.0.x` to have it available in PrestaShop 9.

Reply:
@kpodemski can he target 8.2.x branch or too late ?

Reply:
@kpodemski isUrl already check if string is not empty

Reply:
@jf-viguier I think he can, it's quite a safe pull request, two things to make sure it will go smooth:

- it is important to check that if there is no module hooked, the method behaves like before
- have a simple test module implementing the hook, it could even replace the URL to go to the FoP website ;-)

Reply:
@jf-viguier 

![image](https://github.com/user-attachments/assets/41719a40-4254-407e-a10e-d723e338ac61)

Reply:
i change te code